### PR TITLE
fix(47492): more detailed parseTryStatement error message

### LINKED ISSUE #47492: Missing "catch" after "try" reports "finally expected", rather than noting the lack of a catch
This is totally silly, but I was writing something this weekend, brainfarted, and wrote the following code:

```ts
try {
    
} else {
    
}
```

[Playground Link](https://www.typescriptlang.org/play?ts=4.5.4#code/C4JwngBA3gUB8RgXwgUwDYGdXTg5QA)

It took me _way_ too long to figure out what I did wrong ("why is this squiggling?"), but the error that TS gave me was "'finally' expected", which didn't really push me in the right direction to see that what I actually forgot was to use the word "catch" (which is _way_ more common than `try {} finally {}`).

Maybe this is something that's a byproduct of that it's `try` then `catch` then `finally`, so `finally` is last, but I think I probably would have realized my error a little sooner if the message told me something different, like "catch expected" or "catch or finally expected".

### PR DESCRIPTION
**Pull request by mickey-stringer**
Fixes #47492

Did not add or modify tests, as this simply updates the error message thrown in the invalidTryStatements.ts test

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -6080,7 +6080,14 @@ namespace ts {
             // one out no matter what.
             let finallyBlock: Block | undefined;
             if (!catchClause || token() === SyntaxKind.FinallyKeyword) {
-                parseExpected(SyntaxKind.FinallyKeyword);
+                const catchOrFinallyExpectedMessage: DiagnosticMessage = {
+                    code: 1005,
+                    category: DiagnosticCategory.Error,
+                    key: "_0_expected_1005",
+                    message: "'catch' or 'finally' expected."
+                };
```
Comment (jakebailey) [inline src/compiler/parser.ts:6088]: I don't think it's quite right to do this here; diagnostic messages are generated from `diagnosticMessages.json` so they can be localized, so this probably should be going there, then referenced like other messages.

**Review by jakebailey — CHANGES_REQUESTED** [review]
Oops, meant to click "request changes".

File: src/compiler/parser.ts (hunk 1 above)
Comment (mickey-stringer) [reply src/compiler/parser.ts]: Good point, thanks Jake. I just added a commit moving the new message to diagnosticMessages.json.

**Comment by jakebailey** [comment]
Thanks! This looks about right, though I don't quite know what the rules are for adding new diagnostic messages when it comes to the choice of code. This seems correct to me, at least (the 1000s).

**Comment by mickey-stringer** [comment]
> Thanks! This looks about right, though I don't quite know what the rules are for adding new diagnostic messages when it comes to the choice of code. This seems correct to me, at least (the 1000s).

I missed the notes on picking a code [here](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#diagnostic-messages).
Based on "use the first integral number greater than the last used number in the appropriate range" I think the code should be 1472. Just added another commit with that update.

**Review by DanielRosenwasser — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Pull request merged**